### DISCUSSION TITLE
Additional instructions for offical libre 3 app

### DISCUSSION THREAD

**Discussion by twinko** [maheini/FreeStyle-Libre-3-patch#27] [General]
Hey,

since juggluco 3.x everybody with the possibility to download the offical libre 3 app, doesnt need to download the patched app anymore. This makes the whole setup process much more easy.

In short:
1. download offical app and set up libreView (libreView is important)
2. start a new sensor with the offical libre app
3. Wait for the sensor to send data to libre3 app
4. - Juggluco should retrieve the Libreview account ID: in Juggluco goto left menu->settings->Libreview. Give E-Mail address and password and press "Get Account ID". Leave this view and come back (close little popup window and open again), now there should be a number other then 0 displayed before "Get Account ID". (copied from here: http://jkaltes.byethost16.com/Juggluco/libre3/)
5. Force Stop original libre3 app
6. scan sensor with juggluco
7. Done

I dont have the time right now to right it by myself, perhaps someone else has the time atm.

**Comment by gui-dos** [comment]


**Reply by twinko** [reply]
isnt the solution described above way more easy?

**Reply by twinko** [reply]
Hey this is amazing, i love getting some background info!
But the main purpose of my post was to add some instructions to the tutorial, so everybody can get there libre 3 running with juggluco/xdrip as easily as possible.
that's the reason why i was a little confused. But sure, i love to know the background of things better 😊

**Reply by twinko** [reply]
@gui-dos why did you delete your post? I guess I'm not the only person interested in some more information?!

**Comment by maheini** [comment]
Thank you @twinko 

I'm currently very busy and therefore not able to add any further instructions onto this repo (I have a tested & working manual as a .pdf).... 

This quick & easy instruction is a very good first solution. I pinned it onto the discussions thread.

**Reply by maheini** [reply]
German instructions are out: https://github.com/maheini/FreeStyle-Libre-3-patch/blob/main/Juggluco-solution/juggluco-direct-instructions/de/instructions.md

English version will follow within the next few weeks.

**Comment by MD2240** [comment]
Hello, I just wanted to confirm that this solution is currently working with Libre3 from the USA. I can update as time goes on if any problems arise. I only have this one sensor to test for now(LibreUS14 day is my current RX). 
Setup:
Galaxy S21 on OS 13
Juggluco :
  Version Code: 338
  Version Name: 4.2.7
  Build time: 09:46:23 28-Nov-2022
Libre3USA:
Exp 2023-5-31

Many thanks for all the hard work! 

Matt

**Comment by maheini** [comment]
Wow, those are great news! 🎉 

Thank you for the hint - the instructions will be updated soon.

**Comment by Artem-B** [comment]
I can also confirm that juggluco can read my libre3 (US, Samsung Galaxy Note 10+).

The only oddity I see is that while I see frequent sensor updates on the phone, the data that shows up in libreview has a lot of long gaps between samples. Exported streaming data has all the 1-minute samples, but libreview only has few irregularly spaced points over the same period. Granted, libe3 app seems to have similar problem, so it may be an issue on the libreview side.

### Update on libreview export issue above: 

It does seem to be something funky with libreview export. When it generates the report, the graphs there appear to have all the data. The dots on the chart seem to match the data points exported by libreview, but all the wiggles in the solid chart line appear to match the chart in the juggluco app. 
![image](https://user-images.githubusercontent.com/526795/212604986-e22bda75-ca14-47b6-b325-95a3b8ccf3d9.png)

**Reply by chrisdex** [reply]
Hi Artem,
Is this the only solution that works for the US libre 3 ?

So you start a sensor with the regular app, kill the app and then receive data via jogluco and xdrip / aaps for 2 weeks ?

Thank you !
Cheers
Chris

**Reply by maheini** [reply]
Hi chrisdex

I'm not Artem, but yes, it's the solution which currently works best. There is only one similar alternative I know atm, Diabox app. There's a telegram chat with more information about it.

Your guide is correct. There are further instructions on this repo.

**Reply by j-kaltes** [reply]
> I can also confirm that juggluco can read my libre3 (US, Samsung Galaxy Note 10+).
> 
> The only oddity I see is that while I see frequent sensor updates on the phone, the data that shows up in libreview has a lot of long gaps between samples. Exported streaming data has all the 1-minute samples, but libreview only has few irregularly spaced points over the same period. Granted, libe3 app seems to have similar problem, so it may be an issue on the libreview side.
> ### Update on libreview export issue above:
> 
> It does seem to be something funky with libreview export. When it generates the report, the graphs there appear to have all the data. The dots on the chart seem to match the data points exported by libreview, but all the wiggles in the solid chart line appear to match the chart in the juggluco app. ![image](https://user-images.githubusercontent.com/526795/212604986-e22bda75-ca14-47b6-b325-95a3b8ccf3d9.png)

Juggluco only now and then sends a one minute data point to Libreview. Libreview doesn't use them for statistics. Views and scans are shown in Libreview with an unfilled circle, but who wants the graph to consist only of unfilled circles? Libreview uses the 5 minute (History) data to draw the curve and calculate statistics. If you want to do something with the 1 minute data, use Left middle menu->Export->Stream in Juggluco.

**Comment by master3395** [comment]
A small hint, if you get the yellow warning on the "Added connections", log out of the app, force close it, restart the app, and re-login.
You might have put in a broken ID for the LibreView.

I do not think force-stopping the original Libre3 app is a good solution if you use diasend because juggluco does not have it.

I had to revert back to version 4.4.3 of juggluco to be able to use the patched app (3.3.0.9092)

I tried the latest version of the Libre app and the latest Juggluco app, it did not receive any data at all.

**Comment by Npavkov** [comment]
I have tested the Juggluco app and would choose to use it permanently if someone can provide setup instructions to get Juggluco to connect with xdrip+ (need the xdrip+ setup instructions also). I have tried multiple times to figure out the juggluco settings and the xdrip+ settings however I have not been successful. There are some things I really like about Jugglco like the fact that it sends data easily to Libreview web, and the 1 minute sensor read updates. _**However the things I do not like about Juggluco are * Inconsistent Bluetooth connection with Libre2 sensor*, and  inability to figure out how to connect Juggluco with xdrip+.**_ 
If there is anyone that has successfully setup Juggluco to connect with xdrip+ that can provide the settings needed in both apps to get the connection to function. If there is anyone, I would be well OPEN to trying Juggluco again.

**Reply by jesselockwood** [reply]
This site gives most of the instructions unfortunately they have changed some of the wording in both apps  (https://androidaps.readthedocs.io/lt/latest/Hardware/Libre3.html)
Follow their directions and make the following changes  in the Xdrip+ settings area it says Select "Libre2 patched“ or "Libre 2 (patched app) " My version doesn't have that verbiage I selected  "Libre (patched)" 
When you get to the  Juggluco setting section the instruction says to check the box "Send to Xdrip" newer version doesn't seem to have that verbiage so I check the box that says "Patched Libre Broadcast"  That worked for me. I hope it helps.

**Reply by maheini** [reply]
Thanks @jesselockwood for sumbitting.
An update-instruction is planned, however, I barely got time atm. If there's someone willing to - any PR is appreciated 😄

**Comment by Agonyzyr95** [comment]
Hi, the app has been a literal lifesaver, but the frestyle libre 3 was working for me and then suddenly wouldn't today on a new sensor and it cannot get the libreview id from account info anymore.
![Screenshot_20230918-201933](https://github.com/maheini/FreeStyle-Libre-3-patch/assets/145412468/2c72b54d-56d8-4606-9d3b-d0f5728a478e)
This is the error I've been getting

**Reply by twinko** [reply]
Why do you ask this question in a thread that talks about project updates?
please create a own thread, regarding your problem and add more details, like us or eu libre3, juggluco version and phone. And fill in your login mailadress from. Libre account.

**Reply by j-kaltes** [reply]
I get your error message when I turn off WIFI.

**Comment by Agonyzyr95** [comment]
Thanks for getting back to me so fast, it was my wifi. For some reason the firewall was blocking the server where Libreview data goes and I live out in the country where I don't always have mobile data.
I don't know why its being blocked when it hasn't been before, and I feel kind of dumb. Thank you

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by Agonyzyr95** [comment]
This is the place that the Juggluco site directs for US sensors working with Juggluco and if suddenly it's not, where better to ask the question in case others with the US sensors have similar problems. I did miss some of the potentially relevant info but provided enough info that the dev was able to help me. That was kind of rude for a response imo

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by twinko** [reply]
Open a new discussion here:
https://github.com/maheini/FreeStyle-Libre-3-patch/discussions

The whole repository is for us and eu, always specify about what version you talk about.
